pkg/api/types.go

```diff
@@ -2353,6 +2363,20 @@ type ServiceSpec struct {
 	// cloud-provider does not support the feature."
 	// +optional
 	LoadBalancerSourceRanges []string
+
+	// externalTraffic denotes if this Service desires to route external traffic to
+	// local endpoints only. This preserves Source IP and avoids a second hop for
+	// LoadBalancer and Nodeport type services.
+	// +optional
+	ExternalTraffic ServiceExternalTrafficType
```

Comment:
Sounds good, so the candidates would be:
```
externalTrafficPolicy: Local
externalTrafficPolicy: Global
```

And do we want to use `externalTrafficPolicy: None` for the other service types (not NodePort or LoadBalancer)? This field is now being defaulted to empty string ("").